documentation/intro-to-docs.md

```diff
@@ -0,0 +1,153 @@
+
+### Make documents serve a single purpose
+
+A document that tries to do too many things ends up doing none of them
+well. If you are writing a step-by-step how-to on a specific process,
+avoid going into a deep dive on the involved systems. In addition, keeping
+your documentation focused will mean individual documents will be kept
+shorter, which is important if you want people to be able to read them
+easily and find the information they are looking for.
+
+Consider breaking your documentation into a set of distinct categories,
+and make sure the type of document is very clear in the title (so you
+could title the document "Runbook: Spacecats DB Alerts," for instance).
+Use an ADR to define what these types are -- such as:
+
+* *How-tos* are step-by-step guides to common procedures intended for
+  execution under normal operation. A reader should be able to execute
+  these steps safely when following the document.
+* *Runbooks* are documents intended to be used in concert with alerts
+  that guide troubleshooting and mitigation responses and should include
+  information on issues that users may expect with upstream and downstream
+  services. A reader should be able to perform a quick diagnosis of an
+  alert's underlying causes, or at least rule out any common issues, with
+  the information provided in the document.
+* *Topical guides* provide a deep dive into how a specific system or
+  service works, with references to ADRs that illustrate how technical
+  decisions were made on the project. Readers should come away with a
+  strong understanding of how the system works, as well as how to gather
+  more information about the system and its state on their own.
+
+These aren't meant to be prescriptive; define your categories in a way
+that works best for your project. However, whatever you decide, the
+categories should be clearly defined and focused; notice the "what the
+reader should get from this document" part of the description -- if you
+cannot adequately summarize what a reader is supposed to get from the
+document in a single sentence, it is probably too widely scoped.
+
+### Keep your audience in mind always
+
+The most important thing when writing documentation is to remember that
+you are writing this *for* someone. It may be new engineers, it may
+be people on-call, it may be stakeholders in the C-suite, but whoever they
+are, you intend for *someone* to be reading this at some point in the
+future.
+
+If you aren't keeping this audience in mind throughout the process of writing
+your docs, then you are going to miss the mark. Writing a document intended
+for new engineers and assuming a ton of knowledge will severely limit its
+utility. Similarly, writing something intended for project leads and
+spending pages defining common terms they're already familiar with will
+probably cause them to lose patience and start to skim things over.
+
+### Avoid going overboard with graphics or video
```

Comment:
In the technical writing course there was a tip to write the caption first for any image you wanted to use and make sure that's what the image actually conveys when someone looks at it.